tests/workfiles2_functions.py

```diff
@@ -0,0 +1,196 @@
+    ), "{0} task shouldn't be visible in content dialog".format(entities[2])
+
+    # Remove text in the search field and make sure all tasks are back
+    app_dialog.root["Search All Files"].buttons.mouseClick()
+    assert app_dialog.root.cells[
+        "{0} - {0}".format(entities[2])
+    ].exists(), "{0} task should be visible in content dialog".format(entities[2])
+
+    # Select a specific task
+    app_dialog.root.outlineitems[entities[2]].mouseDoubleClick()
+    app_dialog.root.outlineitems[entities[2]][1].waitExist(timeout=30)
+    app_dialog.root.outlineitems[entities[2]][1].mouseClick()
+
+    # Validate content dialog
+    assert app_dialog.root.cells[
+        "{0} - {0}".format(entities[2])
+    ].exists(), "Not on the right tasks"
+
+    # Validate breadcrumb
+    assert app_dialog.root.captions[
+        "{0}*{1}*{2}*{3}*Step*{4}*Task*{4}".format(
+            tab_name, selection_hierarchy[0], entities[0], entities[1], entities[2]
+        )
+    ].exists(), "Breadcrumb not on the right task"
+
+    # Click on the back navigation button until back to the entity context
+    for _i in range(0, 4):
+        # Click on the back navigation button
+        app_dialog.root.buttons["nav_prev_btn"].mouseClick()
+
+    # Validate Breadcrumb widget is only showing the entity context
+    assert app_dialog.root.captions[
+        tab_name
+    ].exists(), "Breadcrumb widget is not set correctly"
+
+    # Unselect all Pipeline Step filters and make sure tasks are not showing up anymore
+    app_dialog.root.buttons["Select None"].mouseClick()
+    assert (
+        app_dialog.root.outlineitems[entities[2]].exists() is False
+    ), "{0} task shouldn't be visible".format(entities[2])
+    assert (
+        app_dialog.root.cells["{0} - {0}".format(entities[2])].exists() is False
+    ), "{0} task shouldn't be visible in content dialog".format(entities[2])
+
+    # Select all Pipeline Step filters and make sure all tasks are showing up
+    app_dialog.root.buttons["Select All"].mouseClick()
+    app_dialog.root.outlineitems[entities[2]].waitExist(timeout=30)
+    assert app_dialog.root.outlineitems[
+        entities[2]
+    ].exists(), "{0} task should be visible".format(entities[2])
+
+    # Search for a specific task and make sure other tasks are not showing up anymore
+    app_dialog.root["Search Entity"].pasteIn(entities[3], enter=True)
+    assert app_dialog.root.outlineitems[
+        entities[3]
+    ].exists(), "{0} task should be visible".format(entities[3])
+    assert (
+        app_dialog.root.outlineitems[entities[2]].exists() is False
+    ), "{0} task shouldn't be visible".format(entities[2])
+
+    # Enable My Tasks Only and make sure tasks are not showing up anymore
+    app_dialog.root.checkboxes["My Tasks Only"].mouseClick()
+    assert (
+        app_dialog.root.outlineitems[entities[3]].exists() is False
+    ), "{0} task shouldn't be visible".format(entities[3])
```

Comment:
A few things here

As mentioned above the entities parameter is a bit vague maybe that needs to be broken up a bit into separate parameters. I didn't break up the selection hierarchy parameter when I suggested it to you, since the order is important. But I'm not sure we are relying on the order of the entities parameter here, it just looks like we are bundling a bunch of entity names together under a single parameter.

Also not a requirement for merging, but it would be nice if this could be broken up into a few smaller methods if possible and you have time. But only do it if it makes sense to do it.